### PR TITLE
Introducing Remote operation to fetch users' quota

### PR DESCRIPTION
**Pull request by MMMarcy**
As the title said this simpli add a remote user operation.
Depends on https://github.com/owncloud/android-library/pull/38

### CODE REVIEW

File: src/com/owncloud/android/lib/resources/users/GetRemoteUserQuotaOperation.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+
+/**
+ * Created by marcello on 11/11/14.
+ */
+public class GetRemoteUserQuotaOperation extends RemoteOperation {
+
+    private static final String TAG = GetRemoteUserNameOperation.class.getSimpleName();
```
Comment (davivel) [inline src/com/owncloud/android/lib/resources/users/GetRemoteUserQuotaOperation.java]: Typo, getting the name from other class

File: src/com/owncloud/android/lib/resources/users/GetRemoteUserQuotaOperation.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    @Override
+    protected RemoteOperationResult run(OwnCloudClient client) {
+        RemoteOperationResult result = null;
+        int status = -1;
+        GetMethod get = null;
+
+
+        //Get the user
+        try {
+            OwnCloudBasicCredentials credentials = (OwnCloudBasicCredentials) client.getCredentials();
```
Comment (davivel) [inline src/com/owncloud/android/lib/resources/users/GetRemoteUserQuotaOperation.java]: This makes the solution available only for servers with authentication though Basic Auth; need to check the code of OwnCloudCredentials and classes extending it, I won't dare to purpose an alternative from memory.

File: src/com/owncloud/android/lib/resources/users/GetRemoteUserQuotaOperation.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            String url = client.getBaseUri() + OCS_ROUTE + credentials.getUsername();
+
+            get = new GetMethod(url);
+            get.addRequestHeader(createAuthenticationHeader(credentials));
```
Comment (davivel) [inline src/com/owncloud/android/lib/resources/users/GetRemoteUserQuotaOperation.java]: This line not necessary, OwnCloudClient applies the credentials. In this point of the code it should be already authenticated.

File: src/com/owncloud/android/lib/resources/users/GetRemoteUserQuotaOperation.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        }
+
+        return result;
+    }
+
+    private boolean isSuccess(int status) {
+        return (status == HttpStatus.SC_OK);
+    }
+
+    private Header createAuthenticationHeader(OwnCloudCredentials credentials){
```
Comment (davivel) [inline src/com/owncloud/android/lib/resources/users/GetRemoteUserQuotaOperation.java]: This method is not needed.

**Comment by davivel** [comment]
Seen, need some changes. I'll have a look to the issue of getting the username no matter the credentials type.

In any case, I wouldn't make any change until android#737 is reviewed also.

**Comment by davivel** [comment]
owncloud/android#737 was finally reviewed; please take a look at the comments in both

**Comment by malkomich** [comment]
Let's track this PR here: https://github.com/owncloud/android-library/pull/102

**Pull request closed without merging**